### PR TITLE
Continuation of "adds cross-validation for time series" #278

### PR DESCRIPTION
**Pull request by ablaom**
Opening this PR in favour of #278, which has merge conflicts.

### CODE REVIEW

**Comment by CameronBieganek** [comment]
This change

> We didn't address this before, but when `n_obs / (n_folds + 1)` has a remainder, it would probably be nice to put the remainder in the first partition (which is only used for training). Then each test set has exactly the same size.

that I suggested can wait for a later PR. However, this PR is not currently ready to be merged, since it has a bug. The result of the bug is shown in [this comment](https://github.com/alan-turing-institute/MLJBase.jl/pull/278#issuecomment-627442828), and the source of the bug is noted in [this comment](https://github.com/alan-turing-institute/MLJBase.jl/pull/278#discussion_r423864439).

cc: @aa25desh

**Comment by ablaom** [comment]
@aa25desh This PR seems dead. Do you have plans to continue work on it?
cc @vollmersj

**Comment by aa25desh** [comment]
Sorry for late reply, We are working on ND data container for **MLJTime**,  so there would be few more changes to the TimeSeriesCV. @mloning 

> This change
> 
> >that I suggested can wait for a later PR. However, this PR is not currently ready to be merged, since it has a bug. The result of the bug is shown in this comment, and the source of the bug is noted in this comment.
> 

@CameronBieganek So bug would be fixed with only change `n_obs / (n_folds + 1)` instead of `n_obs / (n_folds)` ?

**Comment by ablaom** [comment]
This would be very easy for anyone interested to finish, this PR, which appears to have been abandoned. If someone is interested, I would be happy to resolve the merge conflicts.

**Comment by ablaom** [comment]
@CameronBieganek This PR seems dead, which is a pity. If I resolve the merge conflicts (again), would you be happy to "rescue" it?

**Comment by CameronBieganek** [comment]
@ablaom Yeah, I could probably do that. I don't think there's too much that needs to be done. I need to think a little about whether the data set makes sense. ...I suppose it's probably fine, we would just need to include an example in the docs that shows how you create a data frame with lagged columns and train on that.

**Comment by ablaom** [comment]
Happy to help with moving an example into the docs.

I'll resolve conflicts and ping you when I'm done.

**Comment by CameronBieganek** [comment]
Great, thanks! I will be traveling this weekend, so I won't be able to start right away, but I should be able to look at it soon.

**Comment by CameronBieganek** [comment]
@ablaom I've made some updates to the code, but when I run the MLJBase unit tests on my machine, they're crashing on an error that occurs when testing LossFunctions.jl, which seems unrelated to the time series CV code that I'm working on. My MLJBase environment has LossFunctions.jl v0.7.2. The patch number was bumped 16 days ago. Perhaps there was some change going from 0.7.1 to 0.7.2 that breaks the MLJBase unit tests?

<details>
  <summary>Here are the error messages.</summary>

```
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::DWDMarginLoss{Float64})(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::DWDMarginLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::DWDMarginLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::ExpLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::ExpLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::ExpLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::L1HingeLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::L1HingeLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::L1HingeLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::L2HingeLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::L2HingeLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::L2HingeLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::L2MarginLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::L2MarginLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::L2MarginLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::LogitMarginLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::LogitMarginLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::LogitMarginLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::ModifiedHuberLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::ModifiedHuberLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::ModifiedHuberLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::PerceptronLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::PerceptronLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::PerceptronLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::SigmoidLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::SigmoidLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::SigmoidLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::SmoothedL1HingeLoss{Float64})(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::SmoothedL1HingeLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::SmoothedL1HingeLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - binary: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhatm, ym) ≈ m(yhat, y)
  MethodError: (::ZeroOneLoss)(::UnivariateFiniteVector{OrderedFactor{2}, String, UInt32, Float64}, ::CategoricalVector{String, UInt32, String, CategoricalValue{String, UInt32}, Union{}}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::ZeroOneLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::ZeroOneLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:38 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:11
LossFunctions.jl - continuous: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhat, y) ≈ m(yhat, y)
  MethodError: (::HuberLoss{Float64})(::Vector{Float64}, ::Vector{Float64}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::HuberLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::HuberLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:50
LossFunctions.jl - continuous: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhat, y) ≈ m(yhat, y)
  MethodError: (::L1EpsilonInsLoss{Float64})(::Vector{Float64}, ::Vector{Float64}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::L1EpsilonInsLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::L1EpsilonInsLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:50
LossFunctions.jl - continuous: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhat, y) ≈ m(yhat, y)
  MethodError: (::L2EpsilonInsLoss{Float64})(::Vector{Float64}, ::Vector{Float64}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::L2EpsilonInsLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::L2EpsilonInsLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:50
LossFunctions.jl - continuous: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhat, y) ≈ m(yhat, y)
  MethodError: (::LossFunctions.L2DistLoss)(::Vector{Float64}, ::Vector{Float64}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::LPDistLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::LPDistLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:50
LossFunctions.jl - continuous: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhat, y) ≈ m(yhat, y)
  MethodError: (::LogitDistLoss)(::Vector{Float64}, ::Vector{Float64}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::LogitDistLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::LogitDistLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:50
LossFunctions.jl - continuous: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhat, y) ≈ m(yhat, y)
  MethodError: (::PeriodicLoss{Float64})(::Vector{Float64}, ::Vector{Float64}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::PeriodicLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::PeriodicLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:50
LossFunctions.jl - continuous: Error During Test at /Users/bieganek/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60
  Test threw exception
  Expression: MLJBase.value(m, yhat, X, y, nothing) ≈ LossFunctions.value(m, yhat, y) ≈ m(yhat, y)
  MethodError: (::QuantileLoss{Float64})(::Vector{Float64}, ::Vector{Float64}) is ambiguous. Candidates:
    (loss::LearnBase.SupervisedLoss)(target::AbstractArray, output::AbstractArray) in LossFunctions at /Users/bieganek/.julia/packages/LossFunctions/CKYIO/src/supervised.jl:193
    (m::QuantileLoss)(yhat, y) in MLJBase at /Users/bieganek/projects/MLJBase.jl/src/measures/loss_functions_interface.jl:119
  Possible fix, define
    (::QuantileLoss)(::AbstractArray, ::AbstractArray)
  Stacktrace:
   [1] macro expansion
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:60 [inlined]
   [2] macro expansion
     @ /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [3] top-level scope
     @ ~/projects/MLJBase.jl/test/measures/loss_functions_interface.jl:50
Test Summary:                   | Pass  Error  Total
measures                        |  334     18    352
  aggregation                   |    6             6
  metadata                      |    1             1
  coverage                      |    6             6
  default_measure               |    8             8
  built-in regressor measures   |   13            13
  MLJBase.value                 |    6             6
  built-in classifier measures  |   10            10
  mcr, acc, bacc, mcc           |    6             6
  AUC                           |    5             5
  confusion matrix {2}          |   18            18
  confusion matrix {n}          |   26            26
  Metadata binary               |   41            41
  Metadata multiclass           |   32            32
  More binary metrics           |   27            27
  More multiclass metrics       |   38            38
  Additional multiclass tests   |   22            22
  ROC                           |    2             2
  docstrings coverage           |    1             1
  LossFunctions.jl - binary     |   14     11     25
  LossFunctions.jl - continuous |   14      7     21
  basics                        |   23            23
  ConfusionMatrix measure       |    5             5
ERROR: LoadError: Some tests did not pass: 334 passed, 0 failed, 18 errored, 0 broken.
in expression starting at /Users/bieganek/projects/MLJBase.jl/test/runtests.jl:60
ERROR: Package MLJBase errored during testing
```

</details>

**Comment by CameronBieganek** [comment]
Oh, well at least some of the CI jobs are passing... Not sure what's going on on my machine.

I'm not quite done yet, FYI. I need to edit the docstring. I also haven't really looked at the stuff related to the new sunspot data set.

**Comment by OkonSamuel** [comment]
> Oh, well at least some of the CI jobs are passing... Not sure what's going on on my machine.
> 
> I'm not quite done yet, FYI. I need to edit the docstring. I also haven't really looked at the stuff related to the new sunspot data set.

The ci errors are unrelated to this PR.

**Comment by ablaom** [comment]
@CameronBieganek Thanks for this work. We are waiting on https://github.com/alan-turing-institute/StatisticalTraits.jl/issues/14 , which I am confident will address the 1.0 fail.

Would you like me to review now?

**Comment by CameronBieganek** [comment]
Sure, you can review the code if you want to. However, I need to update the docstring, which I will do as soon as possible...

**Comment by CameronBieganek** [comment]
Oh, and I think I will make one minor, mostly cosmetic change to the `train_test_pairs(::TimeSeriesCV, rows)` method.

**Comment by ablaom** [comment]
Okay, I will wait.

**Comment by ablaom** [comment]
Okay the Julia 1.0 fail is now due to StatisticalTraits 1.1.1 which was yanked from the registry as breaking. 

Bumping [comp] for StatisticalTraits to 2.0 should fix this but must wait for https://github.com/alan-turing-institute/MLJ.jl/issues/810

**Comment by CameronBieganek** [comment]
Ok, @ablaom, this is ready for review.

File: src/resampling.jl [hunk 1]
```diff
@@ -197,6 +201,28 @@ julia> MLJBase.train_test_pairs(TimeSeriesCV(nfolds=3), 1:10)
  (1:4, 5:6)
  (1:6, 7:8)
  (1:8, 9:10)
+
+julia> model = (@load RidgeRegressor pkg=MultivariateStats verbosity=0)();
+
+julia> data = @load_sunspot;
```
Comment (ablaom) [inline src/resampling.jl:207]: I guess this should be `@load_sunspots` (plural) now.

**Comment by ablaom** [comment]
@CameronBieganek Just the one typo in the doc-string.

Many thanks!

**Comment by CameronBieganek** [comment]
Good catch! I fixed the typo. :)

**Comment by ablaom** [comment]
@CameronBieganek Again, many thanks for picking this up! 🙏🏾

**Pull request merged**